aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -116,9 +118,9 @@ struct IndexToOffset {
   const IntArrayRef sizes;
   const IntArrayRef strides;
   const int ndim;
-  IndexToOffset(const Tensor & tensor) : sizes(tensor.sizes()),
-                                         strides(tensor.strides()),
-                                         ndim(tensor.dim()) {}
+  IndexToOffset(const TensorBase & tensor) : sizes(tensor.sizes()),
+                                             strides(tensor.strides()),
+                                             ndim(tensor.dim()) {}
```

Comment:
internal lint is complaining about narrowing from 64-bit to 32-bit here. Make ndim member 64-bit?